lib/facter.rb

```diff
@@ -125,6 +125,26 @@ def debugging(debug_bool)
       Facter::Options[:debug] = debug_bool
     end
 
+    # Iterates over fact names and values
+    #
+    # @yieldparam [String] name the fact name
+    # @yieldparam [String] value the current value of the fact
+    #
+    # @return [void]
```

Comment:
It says that you return void but the method returns self.
What is facter 3 returning?

Reply:
Facter 3 is also returning `Facter`